driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionBypassAutoEncryptionTest.java

```diff
@@ -0,0 +1,125 @@
+import com.mongodb.client.model.Indexes;
+import com.mongodb.client.model.vault.DataKeyOptions;
+import com.mongodb.client.model.vault.EncryptOptions;
+import com.mongodb.client.vault.ClientEncryption;
+import com.mongodb.client.vault.ClientEncryptions;
+import org.bson.BsonBinary;
+import org.bson.BsonString;
+import org.bson.Document;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.security.SecureRandom;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.mongodb.ClusterFixture.isNotAtLeastJava8;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.Fixture.getMongoClient;
+import static com.mongodb.client.Fixture.getMongoClientSettings;
+import static com.mongodb.client.Fixture.getMongoClientSettingsBuilder;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeTrue;
+
+public class ClientSideEncryptionBypassAutoEncryptionTest {
+    private MongoClient clientEncrypted;
+    private ClientEncryption clientEncryption;
+
+    @Before
+    public void setUp() {
+        assumeFalse(isNotAtLeastJava8());
+        assumeTrue(serverVersionAtLeast(4, 1));
+
+        MongoClient mongoClient = getMongoClient();
+
+        final byte[] localMasterKey = new byte[96];
+        new SecureRandom().nextBytes(localMasterKey);
+
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>() {{
+            put("local", new HashMap<String, Object>() {{
+                put("key", localMasterKey);
+            }});
+        }};
+
+        // Set up the key vault for this example
+        MongoNamespace keyVaultNamespace = new MongoNamespace("encryption.testKeyVault");
+        MongoCollection<Document> keyVaultCollection = mongoClient.getDatabase(keyVaultNamespace.getDatabaseName())
+                .getCollection(keyVaultNamespace.getCollectionName());
+        keyVaultCollection.drop();
+
+        // Ensure that two data keys cannot share the same keyAltName.
+        keyVaultCollection.createIndex(Indexes.ascending("keyAltNames"),
+                new IndexOptions().unique(true)
+                        .partialFilterExpression(Filters.exists("keyAltNames")));
+
+        MongoDatabase db = mongoClient.getDatabase(Fixture.getDefaultDatabaseName());
+        db.getCollection("test").drop();
+
+        // Create the ClientEncryption instance
+        ClientEncryptionSettings clientEncryptionSettings = ClientEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(getMongoClientSettings())
+                .keyVaultNamespace(keyVaultNamespace.getFullName())
+                .kmsProviders(kmsProviders)
+                .build();
+
+        clientEncryption = ClientEncryptions.create(clientEncryptionSettings);
+
+        AutoEncryptionSettings autoEncryptionSettings = AutoEncryptionSettings.builder()
```

Comment:
To match what I expect the new prose spec test to say, let's set the mongocryptd spawn path to some garbage value.